moby/moby
# 35012 limit build io

Closes #35012 

**- What we did**
Added the six options to limit read/write speeds to a device that are present in docker run to docker build. There will need to be cli changes to use this feature. We have already completed that work and will open a PR against the cli repo once this gets merged (or earlier if maintainers would prefer that). 

**- How I did it**
Added the additional parameters to the build options, and host config. 

**- How to verify it**
This has been difficult to verify as limiting the block devices is not easily observable for some reason, but you can see that the limits are applied to the build container by checking the cgroup in your docker system files `/sys/fs/cgroup/blkio/docker/<container_long_id>/blkio.throttle.<command-name>`.

**- Description for the changelog**
Add device throttle options to docker build
